File: spring-cloud-openfeign/feign-provider/feign-provider-biz/src/main/java/com/livk/provider/biz/ProviderApp.java
package com.livk.provider.biz;

import com.livk.commons.spring.SpringLauncher;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;

/**
 * <p>
 * ProviderApp
 * </p>
 *
 * @author livk
 * @date 2021/12/6
 */
@EnableCaching
@EnableDiscoveryClient
@SpringBootApplication
public class ProviderApp {

    public static void main(String[] args) {
        SpringLauncher.run(ProviderApp.class, args);
    }

}

File: spring-cloud-openfeign/feign-provider/feign-provider-biz/src/main/kotlin/com/livk/provider/biz/ProviderApp.kt
package com.livk.provider.biz

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cache.annotation.EnableCaching
import org.springframework.cloud.client.discovery.EnableDiscoveryClient

/**
 * @author livk
 */
@EnableCaching
@EnableDiscoveryClient
@SpringBootApplication
open class ProviderApp

fun main(args: Array<String>) {
    runApplication<ProviderApp>(*args)
}
